openstack_plugin_common/__init__.py

```diff
@@ -563,6 +563,11 @@ def _put_client_in_kw(client_name, client_class, kw):
             config = ctx.target.node.properties.get('openstack_config')
     else:
         config = None
+    if 'openstack_config' in kw:
+        if config:
+            config.update(kw['openstack_config'])
```

Comment:
You mean to make a deep copy of config variable before update? And then send this copy (updated of course) to the client_class().get method? I think it is a good idea as well.